ChatTTS/utils/gpu.py

```diff
@@ -34,7 +44,34 @@ def select_device(min_memory=2047, experimental=False):
             logger.get_logger().info("found Apple GPU, but use CPU.")
             device = torch.device("cpu")
     else:
-        logger.get_logger().warning("no GPU found, use CPU instead")
+        logger.get_logger().warning("no GPU or NPU found, use CPU instead")
         device = torch.device("cpu")
 
     return device
+
+
+def _is_torch_npu_available():
+    """
+    check if torch_npu is installed and if a NPU is in the environment
+    """
+    try:
+        import torch_npu
+
+        _ = torch.npu.device_count()  # will raise a RuntimeError if no NPU found
+        return torch.npu.is_available()
+    except (ImportError, RuntimeError):
+        return False
+
+
+def _select_device_if_available(
```

Comment:
Do not wrap it into a function because it is designed just for CUDA. Instead, You can modify the main logic to test whether npu exists first, then CUDA, then MPS.